odoo/models.py

```diff
@@ -2046,6 +2040,15 @@ def _read_group_raw(self, domain, fields, groupby, offset=0, limit=None, orderby
             )
         return result
 
+    def _read_group_resolve_many2one_fields(self, data, fields):
+        many2onefields = [field['field'] for field in fields if field['type'] == 'many2one']
+        for field in many2onefields:
+            ids_set = {d[field] for d in data}
+            m2o_model = self.env[self._fields[field].comodel_name]
+            data_dict = {d[0]: d[1] for d in m2o_model.browse(ids_set).name_get()}
+            for d in data:
+                d[field] = (d[field], data_dict[d[field]]) if d[field] else False
```

Comment:
@odony I opted to keep using `self._fields[field].comodel_name` rather than setting `comodel_name` in  `_read_group_process_groupby`. It made this method less concise as here `many2onefields` holds only the field names (`field['field']`, not all the attributes).

Also had to add `if d[field] else False`  to `d[field] = (d[field], data_dict[d[field]])` as the ORM expects to get `False` where the id is `None` (and not a tuple `(None, u'False')`).